lib/puppet/node.rb

```diff
@@ -108,12 +112,24 @@ def fact_merge
   # Merge any random parameters into our parameter list.
   def merge(params)
     params.each do |name, value|
-      @parameters[name] = value unless @parameters.include?(name)
+      if @parameters.include?(name)
+        Puppet::Util::Warnings.warnonce("The node parameter #{name} for node #{@parameters["hostname"]} was already set to #{@parameters[name]}. It could not be set to #{value}")
```

Comment:
I think there should be single quotes around `#{name}` to make it easier to read. Also noticed that the test did not output anything for the part `#{@parameters["hostname"]}` - maybe that is because of how the test is done. There should be quotes there as well since it looks like the hostname is the empty string.